src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -165,11 +168,10 @@ private ProcessResult tryCreateNewCase(
             .map(awaiting -> awaiting.toString().equals(YesNoFieldValues.YES))
             .orElse(false);
 
-        if (awaitsPaymentProcessing && !ignoreWarnings) {
-            return new ProcessResult(
-                singletonList("Payments for this Exception Record have not been processed yet"),
-                emptyList()
-            );
+        if (awaitsPaymentProcessing && configItem.allowCreatingCaseBeforePaymentsAreProcessed() && !ignoreWarnings) {
+            return new ProcessResult(singletonList(AWAITING_PAYMENTS_MESSAGE), emptyList());
+        } else if (awaitsPaymentProcessing && !configItem.allowCreatingCaseBeforePaymentsAreProcessed()) {
+            return new ProcessResult(emptyList(), singletonList(AWAITING_PAYMENTS_MESSAGE));
```

Comment:
Could we do something like below ?
```
boolean isCaseCreationAllowed = configItem.allowCreatingCaseBeforePaymentsAreProcessed();
             
             if(awaitsPaymentProcessing){
				return isCaseCreationAllowed==true ?
				new ProcessResult(singletonList(AWAITING_PAYMENTS_MESSAGE), emptyList());
				: new ProcessResult(emptyList(), singletonList(AWAITING_PAYMENTS_MESSAGE));
             }

```

Reply:
@nitinprabhu What you suggest excludes ignoreWarnings from the logic

Reply:
boolean isCaseCreationAllowed = 
 configItem.allowCreatingCaseBeforePaymentsAreProcessed();
              if (awaitsPaymentProcessing) {
                 if (!isCaseCreationAllowed) {
                     return new ProcessResult(emptyList(), singletonList(AWAITING_PAYMENTS_MESSAGE));
                 }
                 if (!ignoreWarnings) {
                     return new ProcessResult(singletonList(AWAITING_PAYMENTS_MESSAGE), emptyList());
                 }

Reply:
I **really** want to keep `if-else` style, no hidden exits from the method